src/Analyzers/Core/CodeFixes/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -379,12 +380,13 @@ private async Task<Solution> FixAsync(
             invocationDocument, argument, method.ContainingType, cancellationToken).ConfigureAwait(false);
 
         var newParameterIndex = isNamedArgument ? (int?)null : argumentList.IndexOf(argument);
-        return await AddParameterService.AddParameterAsync(
+        return await AddParameterService.AddParameterAsync<TExpressionSyntax>(
             invocationDocument,
             method,
             argumentType,
             refKind,
-            argumentNameSuggestion,
+            new ParameterName(argumentNameSuggestion, isNamedArgument, tryMakeCamelCase: !method.ContainingType.IsRecord),
+            GetArgument(argument),
```

Comment:
passing more information along to the 'Add Parameter' service so it can attempt to add an appropriate assignment if desired.